src/engine/GroupBy.cpp

```diff
@@ -677,6 +697,126 @@ bool GroupBy::computeOptimizedGroupByIfPossible(IdTable* result) {
   }
 }
 
+// _____________________________________________________________________________
+std::optional<GroupBy::ExplicitlySortedData>
+    GroupBy::checkIfExplicitlySorted() {
+  auto* sort = dynamic_cast<const Sort*>(_subtree->getRootOperation().get());
+  if (!sort) {
+    return std::nullopt;
+  }
+
+  if (_groupByVariables.size() != 1) {
+    return std::nullopt;
+  }
+
+  const Variable& groupByVariable = _groupByVariables.front();
+  auto child = _subtree->getRootOperation()->getChildren().at(0);
+  auto columnIndex = child->getVariableColumn(groupByVariable);
+
+  return ExplicitlySortedData{ columnIndex };
+}
+
+// _____________________________________________________________________________
+template <size_t OUT_WIDTH>
+bool GroupBy::computeGroupByForSortedSubtree(IdTable* result,
+                                             std::vector<Aggregate> aggregates,
+                                             const IdTable& subresult,
+                                             size_t columnIndex,
+                                             LocalVocab& localVocab) {
+  // TODO: Use an interface for AggregationData
+  ad_utility::HashMap<ValueId, AverageAggregationData> map;
+
+  // Initialize evaluation context
+  sparqlExpression::EvaluationContext evaluationContext(
+      *getExecutionContext(), _subtree->getVariableColumns(),
+      subresult,getExecutionContext()->getAllocator(), localVocab);
+
+  evaluationContext._groupedVariables = ad_utility::HashSet<Variable>{
+      _groupByVariables.begin(), _groupByVariables.end()};
+  // TODO: Save previous results (?)
+  evaluationContext._variableToColumnMapPreviousResults =
+      getInternallyVisibleVariableColumns();
+  evaluationContext._previousResultsFromSameGroup.resize(getResultWidth());
+  evaluationContext._isPartOfGroupBy = true;
+
+  size_t blockSize = 65536;
+
+  auto numBlocks = (size_t) ceil(((double) subresult.size()) / blockSize);
+
+  size_t overallResults = 0;
+
+  for (size_t i = 0; i < numBlocks; ++i) {
+    evaluationContext._beginIndex = i * blockSize;
+    evaluationContext._endIndex = std::min((i + 1) * blockSize, subresult.size());
+
+    auto currentBlockSize = evaluationContext._endIndex - evaluationContext._beginIndex;
+
+    std::vector<sparqlExpression::detail::IntOrDouble> resultValues;
+
+    // Is this slower than emplace_back?
+    // Calling IntOrDouble constructor twice VS. resizing of vector on the go
+    resultValues.resize(currentBlockSize);
```

Comment:
If you define the `resultValues` outside the loop, then all but the first one and the last one of these `resize` operations become 0. But I don't know why you need it at all now that I'm thinking about it, see below.